### PR TITLE
[risk=medium][RW-6182] Update publish-cdr for multi-tier

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

Stacked PR onto #4510 which changes configs for multi-tier.

A similar multi-tier PR for the prototype is #4427 

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/db-cdr/generate-cdr/libproject/devstart.rb [hunk 1]
```diff
@@ -8,52 +8,61 @@
...
-    :source_cdr_project => "all-of-us-ehr-dev",
-    :ingest_cdr_project => "fc-aou-vpc-ingest-test",
-    :dest_cdr_project => "fc-aou-cdr-synth-test",
-    :config_json => "config_test.json"
+    :accessTiers => {
+      "registered" => {
+        :source_cdr_project => "all-of-us-ehr-dev",
+        :ingest_cdr_project => "fc-aou-vpc-ingest-test",
+        :dest_cdr_project => "fc-aou-cdr-synth-test",
+        :auth_domain_group_email => "GROUP_all-of-us-registered-test@dev.test.firecloud.org",
```
Comment (jmthibault79) [inline api/db-cdr/generate-cdr/libproject/devstart.rb:16]: PRO: This file contains all the config values it needs
CON: auth_domain_group_email is duplicated in another config file

Alternatively we could update get_auth_domain_group_email() to reference the new config file

File: api/db-cdr/generate-cdr/libproject/devstart.rb [hunk 2]
```diff
@@ -104,7 +120,8 @@ def publish_cdr(cmd_name, args)
   common = Common.new
   env = ENVIRONMENTS[op.opts.project]
   account = env.fetch(:publisher_account)
-  app_sa = "#{ENVIRONMENTS[op.opts.project]}@appspot.gserviceaccount.com"
+  app_sa = "#{op.opts.project}@appspot.gserviceaccount.com"
```
Comment (jmthibault79) [inline api/db-cdr/generate-cdr/libproject/devstart.rb:123]: I think I broke this in #4447 - now it doesn't try to use the whole JSON structure as an email...

File: api/db-cdr/generate-cdr/libproject/devstart.rb [hunk 3]
```diff
@@ -77,6 +86,12 @@ def publish_cdr(cmd_name, args)
     "The Google Cloud project associated with this workbench environment, " +
     "e.g. all-of-us-rw-staging. Required."
   )
+  op.add_option(
+     "--tier [tier]",
+     ->(opts, v) { opts.tier = v},
+     "The access tier associated with this CDR, " +
+     "e.g. registered. Required."
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/libproject/devstart.rb:93]: Instead of making --tier required, can we default it to registered?

**Review by freemabd — APPROVED** [review]
Looks good, just one question about defaulting tier to registered.

File: api/db-cdr/generate-cdr/libproject/devstart.rb (hunk 3 above)
Comment (jmthibault79) [reply api/db-cdr/generate-cdr/libproject/devstart.rb:93]: Is that desirable?  Thinking about the potential of accidentally pushing a CT CDR into the RT here.

Comment (freemabd) [reply api/db-cdr/generate-cdr/libproject/devstart.rb:93]: I don't think there is any harm in having a default tier. Regardless, this command currently won't stop you from publishing a CT CDR into the RT.

Comment (jmthibault79) [reply api/db-cdr/generate-cdr/libproject/devstart.rb:93]: OK, I'll add the default.  https://precisionmedicineinitiative.atlassian.net/browse/RW-6240 for protecting against publishing into the wrong tier

**Pull request merged**